satellite/audit/pathcollector.go

```diff
@@ -9,9 +9,14 @@ import (
 
 	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/metainfo"
 )
 
+var _ metainfo.Observer = (*PathCollector)(nil)
```

Comment:
These things should have a comment because it will be difficult for people to grasp for what they are.

Reply:
This is the idiomatic way to explicitly state that `*PathCollector` implements the interface `metainfo.Observer`. I can add a comment, but I'm wondering whether the better approach would be to make sure all people know how to use it instead.

Reply:
I wanted to use in the past to guarantee that a type always implements a specific interface when such an interface isn't used for the same package. But with some people who I was working didn't like it, but anyway, that's another story.

For me it's OK, maybe the question is if we should always add them otherwise, when?

I thought that you added because the tool that you have created needs a way to identify which interface implements a specific type, but I may have thought wrong.

Reply:
Yes, I added this, because the tool needs some way to detect the depndency.

But, usually, yes, you should add them, unless it causes some sort of dependency cycle. Also for really common interfaces, like `io.Writer`, `io.Closer`, `fmt.Stringer`, it probably doesn't make much sense.

Reply:
Can we add a short comment in some of our current documentation about this?

Reply:
https://github.com/storj/docs/pull/109